packages/toolbox-adk/src/toolbox_adk/credentials.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from google.auth import credentials as google_creds
+
+
+class CredentialType(Enum):
+    TOOLBOX_IDENTITY = "TOOLBOX_IDENTITY"
+    APPLICATION_DEFAULT_CREDENTIALS = "APPLICATION_DEFAULT_CREDENTIALS"
+    USER_IDENTITY = "USER_IDENTITY"
+    MANUAL_TOKEN = "MANUAL_TOKEN"
+    MANUAL_CREDS = "MANUAL_CREDS"
+
+
+@dataclass
+class CredentialConfig:
+    type: CredentialType
+    # For APPLICATION_DEFAULT_CREDENTIALS
+    target_audience: Optional[str] = None
+    # For USER_IDENTITY
+    client_id: Optional[str] = None
+    client_secret: Optional[str] = None
+    scopes: Optional[List[str]] = None
+    # For MANUAL_TOKEN
+    token: Optional[str] = None
+    scheme: Optional[str] = None
+    # For MANUAL_CREDS
+    credentials: Optional[Any] = None  # google.auth.credentials.Credentials
+
+
+class CredentialStrategy:
+    """Factory for creating credential configurations for ToolboxToolset."""
+
+    @staticmethod
+    def TOOLBOX_IDENTITY() -> CredentialConfig:
```

Comment:
Do we need Toolbox Identity? I'm not sure if this is clear what this means for users and if this is a required for the client if it's defaulting to a local auth.